code/game/gamemodes/changeling/evolution_menu.dm

```
@@ -1,300 +1,85 @@
-				</td>
-			</tr>
-			<tr id='search_tr'>
-				<td align='center'>
-					<b>Search:</b> <input type='text' id='filter' value='' style='width:300px;'>
-				</td>
-			</tr>
-	</table>
-
-	"}
-
-	//player table header
-	dat += {"
-		<span id='maintable_data_archive'>
-		<table width='560' align='center' cellspacing='0' cellpadding='5' id='maintable_data'>"}
-
-	var/i = 1
-	for(var/datum/action/changeling/cling_power in GLOB.sting_paths)
-
-		if(cling_power.dna_cost <= 0) //Let's skip the crap we start with. Keeps the evolution menu uncluttered.
-			continue
-
-		var/ownsthis = changeling.has_sting(cling_power)
-
-		var/color
-		if(ownsthis)
-			if(i%2 == 0)
-				color = "#d8ebd8"
-			else
-				color = "#c3dec3"
-		else
-			if(i%2 == 0)
-				color = "#f2f2f2"
-			else
-				color = "#e6e6e6"
-
-
-		dat += {"
-
-			<tr id='data[i]' name='[i]' onClick="addToLocked('item[i]','data[i]','notice_span[i]')">
-				<td align='center' bgcolor='[color]'>
-					<span id='notice_span[i]'></span>
-					<a id='link[i]'
-					onmouseover='expand("item[i]","[cling_power.name]","[cling_power.desc]","[cling_power.helptext]","[cling_power]",[ownsthis])'
-					>
-					<b id='search[i]'>Evolve [cling_power][ownsthis ? " - Purchased" : (cling_power.req_dna>changeling.absorbedcount ? " - Requires [cling_power.req_dna] absorptions" : " - Cost: [cling_power.dna_cost]")]</b>
-					</a>
-					<br><span id='item[i]'></span>
-				</td>
-			</tr>
-
-		"}
-
-		i++
-
-
-	//player table ending
-	dat += {"
-		</table>
-		</span>
-
-		<script type='text/javascript'>
-			var maintable = document.getElementById("maintable_data_archive");
-			var complete_list = maintable.innerHTML;
-		</script>
-	</body></html>
-	"}
-	return dat
-
-
-/datum/action/changeling/evolution_menu/Topic(href, href_list)
-	..()
-	if(!(iscarbon(usr) && usr.mind && usr.mind.changeling))
-		return
+		if("set_view_mode")
+			var/new_view_mode = text2num(params["mode"])
+			ASSERT(new_view_mode in list(COMPACT_MODE, EXPANDED_MODE))
```

I wouldn't use `ASSERT` as the passed value can be altered by the client. If they decided to voluntarily spam the act with an invalid mode, the assertion would fail every time and cause runtimes. Just have it silently fail with a `return`